src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -358,17 +358,18 @@ protected void CheckDocumentIsOpen(DocumentId documentId)
         protected internal void OnDocumentOpened(DocumentId documentId, SourceTextContainer textContainer, bool isCurrentContext = true)
         {
             SetCurrentSolution(
+                data: (@this: this, documentId, textContainer, isCurrentContext),
                 static (oldSolution, data) =>
                 {
-                    var (@this, documentId, _, _) = data;
+                    var (@this, documentId, textContainer, _) = data;
```

Comment:
more consistently tried to deconstruct the data needed and use it, versus a hybrid of accessing some properties as locals and some properties through 'data'.